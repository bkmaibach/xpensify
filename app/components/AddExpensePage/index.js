import React from 'react';
import ExpenseForm from "../ExpenseForm"
import { connect } from 'react-redux';
import { addExpense } from "../../actions/expenses";

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    // props.dispatch(addExpense(expense));
    this.props.addExpense(expense);
    this.props.history.push('/');
  }

  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => dispatch(addExpense(expense))
});

// The second function param is called mapDispatchToProps and is needed because
// of the difficulty in sending a spy in place of addExpense
// (addExpense comes from an imoprt, not a prop)
// It basically turns two(functions(x)) into oneFunction(x)
export default connect(undefined, mapDispatchToProps)(AddExpensePage);
