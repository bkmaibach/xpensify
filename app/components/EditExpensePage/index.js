import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from '../ExpenseForm';
import { editExpense, removeExpense } from "../../actions/expenses";

const EditExpensePage = (props) => (
  <div>
    <ExpenseForm
      expense={props.expense}
      onSubmit={(expense) => {
        props.dispatch(editExpense({id: props.expense.id, updates: expense}));
        props.history.push('/');
        console.log('updated', expense);
      }}
    />
    <button
      onClick={(e) => {
        props.dispatch(removeExpense({id: props.expense.id}));
        props.history.push('/');
      }
    }>Remove</button>
  </div>
  // <div>Editing expense with an ID of: {props.match.params.id}</div>
);

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find( (expense) => expense.id === props.match.params.id)
  };
};

export default connect(mapStateToProps)(EditExpensePage);
