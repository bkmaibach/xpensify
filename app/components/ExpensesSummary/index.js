import React from 'react';
import { connect } from 'react-redux';
import expensesTotalSelector from '../../selectors/expenses-total';
import expensesSelector from '../../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = ({count, total}) => (
  <div>
    {count === 0 ? (
      <h3>Add expenses to begin</h3>
    ) : (
    <h3>Viewing {count} expense{ count > 1 ? ('s') : ('') } totalling {numeral(total/100).format('$0,0.00')}</h3>
    ) }
  </div>
);

const mapStateToProps = (state) => {
  const selectedExpenses = expensesSelector(state.expenses, state.filters);
  return {
    total: expensesTotalSelector(selectedExpenses),
    count: selectedExpenses.length
  }
}

export default connect(mapStateToProps)(ExpensesSummary);