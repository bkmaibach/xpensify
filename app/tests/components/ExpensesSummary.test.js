import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render correctly with no expenses', () => {
  const wrapper = shallow(<ExpensesSummary count={0} expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render correctly with one expense', () => {
  const wrapper = shallow(<ExpensesSummary count={1} expenses={[expenses[2]]}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render correctly with expenses', () => {
  const wrapper = shallow(<ExpensesSummary count={4} expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});