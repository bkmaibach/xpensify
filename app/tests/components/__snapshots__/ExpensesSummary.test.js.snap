// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with expenses 1`] = `
<div>
  <h3>
    Viewing 
    4
     expense
    s
     totalling 
    $0.00
  </h3>
</div>
`;

exports[`should render correctly with no expenses 1`] = `
<div>
  <h3>
    Add expenses to begin
  </h3>
</div>
`;

exports[`should render correctly with one expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     expense
     totalling 
    $0.00
  </h3>
</div>
`;
